core/modules/util/Timer.cc

```diff
@@ -58,4 +67,65 @@ std::ostream & operator<<(std::ostream & os, Timer const & timer) {
     return os;
 }
 
+
+LockGuardTimed::LockGuardTimed(std::mutex& mtx, std::string const& note)
+    : _mtx(mtx), _note(note) {
+    timeToLock.start();
+    _mtx.lock();
+    timeToLock.stop();
+    timeHeld.start();
+}
+
+LockGuardTimed::~LockGuardTimed() {
+    _mtx.unlock();
+    timeHeld.stop();
+    LOGS(_log, LOG_LVL_DEBUG, "lockTime " << _note << " toLock=" << timeToLock.getElapsed() <<
+                              " held=" << timeHeld.getElapsed());
+}
+
```

Comment:
You might want to consider simply doing a timed mutex. Then you can use it in more than just a "gaurd" context.

Reply:
The point of this is just have something easy to drop into the code to replace a regular lock_guard and see how long it is taking to acquire the lock and how long it is held. Trying to add other functionality (say like that of unique_lock) made it get much more complicated. I'd rather leave it at this until there's a demonstrated need.